File: OPAL/tac/src/main/scala/org/opalj/tac/fpcf/analyses/string/interpretation/SoundnessMode.scala
/* BSD 2-Clause License - see OPAL/LICENSE for details. */
package org.opalj
package tac
package fpcf
package analyses
package string
package interpretation

trait SoundnessMode {

    def isHigh: Boolean
}

object SoundnessMode {

    def apply(high: Boolean): SoundnessMode = if (high) HighSoundness else LowSoundness
}

object LowSoundness extends SoundnessMode {

    override def isHigh: Boolean = false
}

object HighSoundness extends SoundnessMode {

    override def isHigh: Boolean = true
}

File: DEVELOPING_OPAL/validate/src/test/java/org/opalj/fpcf/properties/string/SoundnessMode.java
/* BSD 2-Clause License - see OPAL/LICENSE for details. */
package org.opalj.fpcf.properties.string;

/**
 * Restricts an annotation to certain soundness mode configurations.
 * @see org.opalj.fpcf.fixtures.string.SimpleStringOps
 *
 * @author Maximilian Rüsch
 */
public enum SoundnessMode {

    LOW("LOW"),
    HIGH("HIGH");

    private final String value;

    SoundnessMode(String value) {
        this.value = value;
    }

    public String getValue() {
        return value;
    }
}
